### ISSUE TITLE
Type generation on new project gives inconsistent results

### ISSUE THREAD

**Issue by fnimick** [supabase/cli#1165]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

Running `supabase gen types typescript --local` as provided in the CI example at https://github.com/supabase/supabase-action-example/blob/main/.github/workflows/ci.yaml does not work on a brand new project. The types are substantially different when generated locally vs on the CI runner.

```
@@ -53,36 +53,24 @@ export interface Database {
     Tables: {
       buckets: {
         Row: {
-          allowed_mime_types: string[] | null
-          avif_autodetection: boolean | null
           created_at: string | null
-          file_size_limit: number | null
           id: string
           name: string
           owner: string | null
-          public: boolean | null
           updated_at: string | null
         }
         Insert: {
-          allowed_mime_types?: string[] | null
-          avif_autodetection?: boolean | null
           created_at?: string | null
-          file_size_limit?: number | null
           id: string
           name: string
           owner?: string | null
-          public?: boolean | null
           updated_at?: string | null
         }
         Update: {
-          allowed_mime_types?: string[] | null
-          avif_autodetection?: boolean | null
           created_at?: string | null
-          file_size_limit?: number | null
           id?: string
           name?: string
           owner?: string | null
-          public?: boolean | null
           updated_at?: string | null
         }
       }
@@ -115,9 +103,7 @@ export interface Database {
           metadata: Json | null
           name: string | null
           owner: string | null
-          path_tokens: string[] | null
           updated_at: string | null
-          version: string | null
         }
         Insert: {
           bucket_id?: string | null
@@ -127,9 +113,7 @@ export interface Database {
           metadata?: Json | null
           name?: string | null
           owner?: string | null
-          path_tokens?: string[] | null
           updated_at?: string | null
-          version?: string | null
         }
         Update: {
           bucket_id?: string | null
@@ -139,9 +123,7 @@ export interface Database {
           metadata?: Json | null
           name?: string | null
           owner?: string | null
-          path_tokens?: string[] | null
           updated_at?: string | null
-          version?: string | null
         }
       }
     }
@@ -149,15 +131,6 @@ export interface Database {
       [_ in never]: never
     }
     Functions: {
-      can_insert_object: {
-        Args: {
-          bucketid: string
-          name: string
-          owner: string
-          metadata: Json
-        }
-        Returns: undefined
-      }
       extension: {
         Args: {
           name: string
@@ -1[76](https://github.com/millionmarker/future/actions/runs/5084948133/jobs/9137881007#step:8:77),13 +149,6 @@ export interface Database {
         }
         Returns: string[]
       }
-      get_size_by_bucket: {
-        Args: Record<PropertyKey, never>
-        Returns: {
-          size: number
-          bucket_id: string
-        }[]
-      }
       search: {
         Args: {
           prefix: string
@@ -1[90](https://github.com/millionmarker/future/actions/runs/5084948133/jobs/9137881007#step:8:91),9 +156,6 @@ export interface Database {
           limits?: number
           levels?: number
           offsets?: number
-          search?: string
-          sortcolumn?: string
-          sortorder?: string
         }
         Returns: {
           name: string
```

## To Reproduce

Create a brand new Supabase project, set up a local project, run `supabase db remote commit`, then `supabase gen types typescript --local`. These generated types will then fail the CI test as provided.

## Expected behavior

The generated types should be identical.

**Comment by fnimick** [comment]
I've pinned the CI cli version to my local cli version as suggested in https://github.com/supabase/cli/issues/1146 , with the same error. With both versions set to 1.63.1, the output is identical.

**Comment by Eldynn** [comment]
@fnimick did you find a way to do that and validate that typing are updated when new PR popup via workflows?

**Comment by soedirgo** [comment]
@fnimick are you also using `supabase gen types typescript --local` when initially generating the types? It looks like you're generating the `storage` schema, and the local dev setup's `storage` schema usually lags slightly behind what's on the platform. If you did `supabase gen types typescript --project-id|--linked` to generate the types at the start, the generated types will differ.

Alternatively, you can omit the `storage` schema and only generate types for schemas you manage, e.g. `supabase gen types typescript --local --schema=public --schema=private ...`.

**Comment by fnimick** [comment]
@Eldynn  not sure exactly what you are asking - I pinned the Supabase CLI version in the CI runner to exactly match my local install, and saw the same error diffs.

@soedirgo in my case, I am running generation against my local setup, which I use for all development. That's why I was expecting a similar local setup with `supabase start` in CI to then give the same result.

I could generate and commit type schemas for only schemas I manage, but then the type information will be missing from typed API calls, would it not? We'd lose type safety when operating the supabase-managed schemas in that case. Unless there is a reason to only generate our managed schemas, in which case the examples and docs should probably be updated.

**Comment by 0xBigBoss** [comment]
I am seeing this as well, but with the citext extension. It's literally just different order running it on mac vs ubuntu, same cli version. `1.87.4`

<img width="1100" alt="Screenshot 2023-08-21 at 22 38 44" src="https://github.com/supabase/cli/assets/95193764/1c73a6d0-401c-4933-ba36-eb997551cc6b">

Which is sad because I'd like to include this as part of the CI process as recommended here, https://supabase.com/docs/guides/cli/managing-environments#configure-github-actions.

**Comment by awalias** [comment]
is this one fixed?

**Comment by 0xBigBoss** [comment]
This is fixed for me in in npm version supabase 1.111.2

**Comment by KevinBrolly** [comment]
Closing this issue as fixed.
